### PR TITLE
8 post new comment

### PR DESCRIPTION
**Pull request by andmikpar**

### CODE REVIEW

File: src/components/ArticleComments.jsx [hunk 1]
```diff
@@ -21,31 +31,98 @@ const ArticleComment = ({ isLoading, setIsLoading, setIsError }) => {
...
+                </button>
+              ) : (
+                <form className="addComment">
+                  <input
+                    onChange={(e) => setInput(e.target.value)}
+                    typeof="text"
+                    value={input}
+                    placeholder="add comment"
+                    className="commentInput"
+                  ></input>
```
Comment (popatre) [inline src/components/ArticleComments.jsx]: Could you change this is a `textarea` to allow for multi line comments?

**Review by popatre — CHANGES_REQUESTED** [review]
Some nice code in here, there's just a few things id change from a UX point of view
- [x] It would be better if the error maybe just appeared below or instead of the post box - at the moment it seems to clear the whole page, just to display the error
- [ ] If you're going to stick with the current system, the back button takes you back to the page before the one you were on before, which is just a bit annoying to have to click comments again.
- [x] It wasnt completely obvious that my comment wasnt posting because i wasnt logged in. Status 400 wasn't helpful and would be easier just to tell the user they need to be logged in to post. It would possibility be even better to render that above message, instead of the button, if someone isnt logged in. That way, they cant even get to the point of posting. 
- [x] when you post a comment - ideally needs a message to say posted successfully , instead of just a refresh button

**Review by Pipe-Smoking-Rabbit — CHANGES_REQUESTED** [review]
Nice work Andy, I'm really enjoying seeing the styling come together on your app, it's very visually pleasant! 👍 
I have some notes for you on this ticket:
- I would prefer to see a newly posted comment render optimistically rather than needing to refresh the page, we live in a very modern era where users expect instant gratification, I would recommend refactoring to use `state` to update the comments list instantly, then run the `post` request in the background 
- Error handling wise, when I switch to offline mode and attempt to post a comment, I still see the following message `"Message posted."`, even though the request has failed! take advantage of the `post` request's `.catch` block to handle for any errors that prevent the request from running successfully and inform the user that something has gone wrong

**Review by popatre — CHANGES_REQUESTED** [review]
Some good changes implemented. However, the UX is still a bit annoying. 

When i post offline, it replaces the whole page with the error message, comments as well. 
You've already changed it so an error message can appear below the box, so i'd just reuse that one. 
If you change the above, this wont be an issue, but it gives you an option to return to the home page. To then post a comment, you have to find which article you were on, and open the comments again. If it kept failing, this cycle would be really annoying for an end user. 

I would just re-use your error message, the one that appears when you try and post when not signed in, and get that to say 'something went wrong.' Means you can still leave the comments on the page, and if they fix the issue, they dont have to go searching around for the article again, they're already in the right place

File: src/components/ArticleComments.jsx [hunk 2]
```diff
@@ -19,15 +29,87 @@ const ArticleComment = ({ isLoading, setIsLoading, setIsError }) => {
...
+    performGet();
+    return postError;
+  }
+
+  function postComment(input) {
+    setPostError(null);
+    if (!input) {
+      setPostError('Empty comments not allowed!');
+    } else if (loggedInUser === 'Sign In') {
+      setPostError('You need to sign in to comment!');
```
Comment (popatre) [inline src/components/ArticleComments.jsx:60]: Re use this setPostError for network errors too!

**Review by popatre — CHANGES_REQUESTED** [review]

**Pull request merged**